pkg/client/core/service_application_client.go

```diff
@@ -86,14 +87,120 @@ func (s *applicationService) ScaffoldApplication(ctx context.Context, opts *clie
+}
+
+func getClusterApplications(fs *afero.Afero, absoluteRepositoryRootDir string, cluster v1alpha1.Cluster) ([]string, error) {
+	absoluteApplicationsDir := path.Join(absoluteRepositoryRootDir, paths.GetRelativeArgoCDApplicationsDir(cluster))
+
+	applications := make([]string, 0)
+
+	items, err := fs.ReadDir(absoluteApplicationsDir)
+	if err != nil {
+		if stderrors.Is(err, os.ErrNotExist) {
+			return applications, nil
+		}
+
+		return nil, fmt.Errorf("listing applications directory: %w", err)
+	}
+
+	for _, item := range items {
+		valid, err := argocd.IsArgoCDApplication(fs, path.Join(absoluteApplicationsDir, item.Name()))
+		if err != nil {
+			return nil, fmt.Errorf("checking if ArgoCD application: %w", err)
+		}
+
+		if valid {
+			applications = append(applications, strings.Replace(item.Name(), path.Ext(item.Name()), "", 1))
+		}
+	}
+
+	return applications, nil
+}
+
+// AmountAssociatedClusters knows how to count the number of associated clusters a specific application has. A cluster is
+// classified as associated iff the cluster has an ArgoCD application referencing the app and the app has an overlay folder
+// referencing the cluster.
+func AmountAssociatedClusters(fs *afero.Afero, absoluteRepositoryRootDir string, clusterContext v1alpha1.Cluster, app v1alpha1.Application) (int, error) {
+	absoluteApplicationDir := path.Join(absoluteRepositoryRootDir, paths.GetRelativeApplicationDir(clusterContext, app))
+
+	clusters, err := getApplicationOverlayClusters(fs, absoluteApplicationDir)
+	if err != nil {
+		return 0, fmt.Errorf("retrieving clusters in overlay directory: %w", err)
+	}
+
+	associatedClusters := 0
+
+	for _, cluster := range clusters {
+		clusterApplications, err := getClusterApplications(fs, absoluteRepositoryRootDir, v1alpha1.Cluster{
+			Metadata: v1alpha1.ClusterMeta{Name: cluster},
+			Github:   v1alpha1.ClusterGithub{OutputPath: clusterContext.Github.OutputPath},
+		})
+		if err != nil {
+			return 0, fmt.Errorf("listing cluster applications: %w", err)
+		}
+
+		if contains(clusterApplications, app.Metadata.Name) {
+			associatedClusters++
+		}
+	}
+
+	return associatedClusters, nil
+}
+
 // DeleteApplicationManifests removes manifests related to an application
 func (s *applicationService) DeleteApplicationManifests(_ context.Context, opts client.DeleteApplicationManifestsOpts) error {
-	absoluteApplicationDir := path.Join(
+	absoluteApplicationOverlaysDirectory := path.Join(
+		s.absoluteRepositoryDir,
+		getRelativeOverlayDirectory(opts.Cluster, opts.Application),
+	)
+
+	err := s.fs.RemoveAll(absoluteApplicationOverlaysDirectory)
+	if err != nil {
+		return fmt.Errorf("deleting overlay directory: %w", err)
+	}
+
+	associatedClusters, err := AmountAssociatedClusters(s.fs, s.absoluteRepositoryDir, opts.Cluster, opts.Application)
+	if err != nil {
+		return fmt.Errorf("counting associated clusters: %w", err)
+	}
+
+	if associatedClusters != 0 {
+		return nil
+	}
+
+	absoluteApplicationRootDirectory := path.Join(
```

Comment:
I find I have to go through a lot of variables and function calls when I'm trying to figure out what this variable really contains. Could we perhaps add a comment that gives an example of what this variable contains? For instance:

```go
// Example: /home/me/my-iac-repo/infrastructure/applications/my-app
```

If so, consider above line 179 as well.

One counter argument might be that it we don't do it other places in the code.